modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/ObjectEntry1toMObjectRelatedModelsProviderImpl.java

```diff
@@ -68,14 +76,67 @@ public void deleteRelatedModel(
 				deletionType,
 				ObjectRelationshipConstants.DELETION_TYPE_CASCADE)) {
 
+			int maxpoolsize = GetterUtil.getInteger(
+				SystemProperties.get("system.executor.service.maxpoolsize"), 2);
+
+			ExecutorService executorService = new ThreadPoolExecutor(
```

Comment:
@pedro-oliveira446 executorService Should be reused rather than creating each time